ts/tests/helpers/skeleton-app.ts

```diff
@@ -0,0 +1,133 @@
+import fs from 'fs-extra';
+import path from 'path';
+import tmp from 'tmp';
+import execa from 'execa';
+import { EventEmitter } from 'events';
+import got from 'got';
+import debugLib from 'debug';
+
+tmp.setGracefulCleanup();
+
+const debug = debugLib('skeleton-app');
+
+const getEmberPort = (() => {
+  let lastPort = 4210;
+  return () => lastPort++;
+})();
+
+export default class SkeletonApp {
+  port = getEmberPort();
+  watched: WatchedBuild | null = null;
+  tmpDir = tmp.dirSync({
+    tries: 10,
+    unsafeCleanup: true,
+    dir: process.cwd(),
+    template: 'test-skeleton-app-XXXXXX',
+  });
+  root = this.tmpDir.name;
+
+  constructor() {
+    fs.copySync(`${__dirname}/../../../test-fixtures/skeleton-app`, this.root);
+  }
+
+  build() {
+    return this._ember(['build']);
+  }
+
+  serve() {
+    if (this.watched) {
+      throw new Error('Already serving');
+    }
+    return (this.watched = new WatchedBuild(
+      this._ember(['serve', '--port', `${this.port}`]),
+      this.port
+    ));
+  }
+
+  updatePackageJSON(callback: (arg: any) => string) {
+    let pkgPath = `${this.root}/package.json`;
+    let pkg = fs.readJSONSync(pkgPath);
+    fs.writeJSONSync(pkgPath, callback(pkg) || pkg, { spaces: 2 });
+  }
+
+  writeFile(filePath: string, contents: string) {
+    let fullPath = `${this.root}/${filePath}`;
+    fs.ensureDirSync(path.dirname(fullPath));
+    fs.writeFileSync(fullPath, contents, 'utf-8');
+  }
+
+  readFile(path: string) {
+    return fs.readFileSync(`${this.root}/${path}`, 'utf-8');
+  }
+
+  removeFile(path: string) {
+    return fs.unlinkSync(`${this.root}/${path}`);
+  }
+
+  teardown() {
+    if (this.watched) {
+      this.watched.kill();
+    }
+    this.tmpDir.removeCallback();
+  }
+
+  _ember(args: string[]) {
+    let ember = require.resolve('ember-cli/bin/ember');
+    return execa('node', [ember].concat(args), { cwd: this.root, cleanup: true });
```

Comment:
Potentially important change, `, cleanup: true` option ensures that killing the process also kills any child processes that it has spawned.

Reply:
I think this option already [defaults to `true`](https://github.com/sindresorhus/execa#cleanup)

Reply:
good catch!